### PR TITLE
small-scale refresh for the sandpile file

### PR DESCRIPTION
**Pull request by fchapoton**
fix some pep8, ruff, mypy warnings

break some of the many long lines

minor changes in the code

### :memo: Checklist

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.

### CODE REVIEW

File: src/sage/sandpiles/sandpile.py [hunk 1]
```diff
@@ -684,7 +688,7 @@ def __getattr__(self, name):
             elif name == '_in_degrees':
                 self._set_in_degrees()
                 return deepcopy(self.__dict__[name])
-            elif name == '_burning_config' or name == '_burning_script':
+            elif name in ['_burning_config', '_burning_script']:
```
Comment (dcoudert) [inline src/sage/sandpiles/sandpile.py]: is it better to use list or tuple in such case ?

File: src/sage/sandpiles/sandpile.py [hunk 2]
```diff
@@ -726,7 +730,7 @@ def __getattr__(self, name):
             elif name in ['_postulation', '_h_vector', '_hilbert_function']:
                 self._set_hilbert_function()
                 return deepcopy(self.__dict__[name])
-            elif (name == '_ring' or name == '_unsaturated_ideal'):
+            elif name in ['_ring', '_unsaturated_ideal']:
```
Comment (dcoudert) [inline src/sage/sandpiles/sandpile.py]: use `if` ? and remove last `else`

File: src/sage/sandpiles/sandpile.py [hunk 3]
```diff
@@ -2080,7 +2101,8 @@ def stable_configs(self, smax=None):
         else:
             c = SandpileConfig(self, smax)
             if not c <= self.max_stable():
```
Comment (dcoudert) [inline src/sage/sandpiles/sandpile.py]: `if c > self.max_stable():` ?

File: src/sage/sandpiles/sandpile.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/sandpiles/sandpile.py]: maybe list is faster to create, but one could also use set. It does not matter much

Comment (dcoudert) [reply src/sage/sandpiles/sandpile.py]: I was just curious between list and tuple. I assume set is longer to create.

**Review by dcoudert — APPROVED** [review]
LGTM.

**Pull request merged**